File: core-api/pkg/handler/schema/account.go
package schema

import "github.com/gin-gonic/gin"

type CreateAccountRequest struct {
	Username string `json:"username"`
	Password string `json:"password"`
}

type LoginAccountRequest struct {
	Username string `json:"username"`
	Password string `json:"password"`
}

type Account struct {
	Id        string `json:"id"`
	Username  string `json:"username"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
}

type TokenWithAccount struct {
	Token   string   `json:"token"`
	Account *Account `json:"account"`
}

type RegisterFcmTokenRequest struct {
	FcmToken string `json:"fcm_token"`
}

type AccountController interface {
	CreateAccount(c *gin.Context)
	LoginAccount(c *gin.Context)
	VerifyToken(c *gin.Context)
	RegisterFcmToken(c *gin.Context)
}

File: client/lib/schema/account.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'account.freezed.dart';

part 'account.g.dart';

@freezed
class CreateAccountRequest with _$CreateAccountRequest {
  const factory CreateAccountRequest({
    required String username,
    required String password,
  }) = _CreateAccountRequest;

  factory CreateAccountRequest.fromJson(Map<String, dynamic> json) =>
      _$CreateAccountRequestFromJson(json);
}

@freezed
class LoginAccountRequest with _$LoginAccountRequest {
  const factory LoginAccountRequest({
    required String username,
    required String password,
  }) = _LoginAccountRequest;

  factory LoginAccountRequest.fromJson(Map<String, dynamic> json) =>
      _$LoginAccountRequestFromJson(json);
}

@freezed
class Account with _$Account {
  const factory Account({
    required String id,
    required String username,
    @JsonKey(name: "created_at") required String createdAt,
    @JsonKey(name: "updated_at") required String updatedAt,
  }) = _Account;

  factory Account.fromJson(Map<String, dynamic> json) =>
      _$AccountFromJson(json);
}

@freezed
class TokenWithAccount with _$TokenWithAccount {
  const factory TokenWithAccount(
      {required String token, required Account account}) = _TokenWithAccount;

  factory TokenWithAccount.fromJson(Map<String, dynamic> json) =>
      _$TokenWithAccountFromJson(json);
}
